src/Data/Graph.purs

```diff
@@ -1,152 +1,107 @@
 -- | A data structure and functions for graphs
 
 module Data.Graph
-  ( Edge(..)
-  , Graph(..)
-  , SCC(..)
+  ( Graph
```

Comment:
Ok, I guess that makes sense since this is going to keep evolving. It might be good to provide a `fromMap` and/or `fromEdges` constructor(s) then to make it convenient to construct from other common graph-y representations?